organizations/models.py

```diff
@@ -29,10 +40,18 @@ def _generate_unique_schema_name(self):
             schema_name = self._generate_random_string()
         return schema_name
 
+    def _generate_unique_api_key(self):
+        """Generates a unique api_key."""
+        api_key = self._generate_random_cryptographically_secure_string()
+        while Organization.objects.filter(api_key=api_key).exists():
+            api_key = self._generate_random_cryptographically_secure_string()
+        return api_key
+
     def save(self, *args, **kwargs):
         """Organization save method. Before checking it creates a unique schema name if does not exist already."""
         if not self.schema_name:
             self.schema_name = self._generate_unique_schema_name()
+            self.api_key = self._generate_unique_api_key()
```

Comment:
this should be preceeded by `if not self.api_key` - we should not mix it with schema name, no?